### PR TITLE
[docker] Fall back to default route if docker hostname is not resolvable

### PR DESCRIPTION
**Pull request by remh**
### What does this PR do?
In case, the agent cannot resolve the docker host hostname, we fall back on the default gateway.

### Motivation

Customer ticket on kubernetes where the agent was unable to resolve the docker host hostname.

### CODE REVIEW

**Comment by remh** [comment]
@truthbk @hkaj to be merged before https://github.com/DataDog/dd-agent/pull/2702 (as it's merging into Haissam's branch)

**Comment by truthbk** [comment]
Related: #2745

File: utils/dockerutil.py [hunk 1]
```diff
@@ -138,11 +140,26 @@ def _get_checks_from_inspect(self, inspect):
...
         except Exception:
-            log.critical("Unable to find docker host hostname")
+            log.critical("Unable to find docker host hostname. Trying default route")
+
+        try:
+            with open('/proc/net/route') as f:
+                for line in f.readlines():
+                    fields = line.strip().split()
+                    if fields[1] == '00000000':
+                        return socket.inet_ntoa(struct.pack('<L', int(fields[2], 16)))
```
Comment (truthbk) [inline utils/dockerutil.py]: We might need some fallback for ipv6 only systems...

**Comment by truthbk** [comment]
Other than the ipv6 concern, this looks good to me.

File: utils/dockerutil.py (hunk 1 above)
Comment (remh) [inline utils/dockerutil.py]: This is to be run within a Docker container which doesn't support ipv6 only networks

**Pull request merged**